src/main/java/tc/oc/pgm/map/MapTagImpl.java

```diff
@@ -0,0 +1,57 @@
+package tc.oc.pgm.map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import tc.oc.pgm.api.map.MapTag;
+
+public class MapTagImpl implements MapTag {
+
+  private static final String HASHTAG = "#";
+  private final String id;
+  private final String name;
+  private final boolean gamemode;
+  private final boolean auxiliary;
+
+  public MapTagImpl(String id, String name, boolean gamemode, boolean auxiliary) {
+    this.id = checkNotNull(id);
```

Comment:
We lost the regex pattern to validate if the maptag was in lower case, between a-z, 0-9, with - or _. This public and people from outside can create new instances of this class with illegal characters which will potentially lead to bugs.